maestro-client/src/main/java/maestro/Driver.kt

```diff
@@ -89,4 +89,6 @@ interface Driver {
     fun waitForAppToSettle(initialHierarchy: ViewHierarchy?): ViewHierarchy?
 
     fun capabilities(): List<Capability>
+
+    fun setPermissions(appId: String, permissions: Map<String, String>)
```

Comment:
What are all possible values for permissions, since it could be different for iOS and android?

Reply:
The values are platform specific. On iOS the list is short and mostly static.

On android the list is quite big, which also makes the list more dynamic. Also I believe that apps on Android can only ask for permissions that they've listed in their manifest, which we can read out with adb. This means we can make "all permissions" mean "all requested permissions" on android, which avoids hard-coding a separate list of "all permissions" that exist on Android.

Reply:
Also, I think the current implementation of `adb pm grant -g <appi>` does exactly that.

Reply:
Aha, that's unfortunate.

Reply:
This PR is going to fix it: https://github.com/mobile-dev-inc/maestro/pull/864